### PR TITLE
setting-up-ipsinline-for-linux: alternative setup with mangle PREROUTING

### PR DESCRIPTION
**Pull request by selivan**
Added alternative iptables option with mangle PREROUTING table. And small fix to easier check for NFQ support in Suricata.

### CODE REVIEW

**Comment by inliniac** [comment]
Thanks!

@regit does this make sense to you?

File: doc/userguide/setting-up-ipsinline-for-linux.rst [hunk 1]
```diff
@@ -117,6 +117,14 @@ The options -i (input) -o (output) can be combined with all previous mentioned o
 
 If you would stop Suricata and use internet, the traffic will not come through. To make internet work correctly, you have to erase all iptable rules.
 
+To simply inspect all traffic on the host, before routing desicion and input/output separation, you can use mangle PREROUTING table:
```
Comment (regit) [inline doc/userguide/setting-up-ipsinline-for-linux.rst]: Typo on decision

File: doc/userguide/setting-up-ipsinline-for-linux.rst [hunk 2]
```diff
@@ -117,6 +117,14 @@ The options -i (input) -o (output) can be combined with all previous mentioned o
...
+
+::
+
+
+  sudo iptables -t mangle -I PREROUTING -j NFQUEUE
```
Comment (regit) [inline doc/userguide/setting-up-ipsinline-for-linux.rst]: This is wrong: this setup only works for FORWARDED traffic. In the case of INPUT/OUTPUT we will only get the INPUT traffic. Look at the output of iptables after doing a SSH from the local box to an external host.
```
iptables -L -nv -t mangle
Chain PREROUTING (policy ACCEPT 273 packets, 287K bytes)
 pkts bytes target     prot opt in     out     source               destination         
   18  4627            tcp  --  *      *       0.0.0.0/0            0.0.0.0/0            tcp spt:22
    0     0            tcp  --  *      *       0.0.0.0/0            0.0.0.0/0            tcp dpt:22
``` 
We don't see the packet in the output direction.

**Comment by selivan** [comment]
@redit, oops, you are right. Fixed that: 0dd3db193246b8d2ffc1a36a06540d5ed2599360.

Anyway, I think sometimes using mangle PREROUTING chain makes sense: if Suricata is running in IPS mode, dropping packet on earlier processing stage is better.

**Comment by inliniac** [comment]
Would it make sense to make this part of a more advanced nfq guide instead? The guide here is really focused at beginners.

Regardless, please don't update existing PRs https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Github_work_flow#Create-a-new-branch-for-incorporating-feedback

Also, having a PR with commits that fix commits in the PR is bad practice. See https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Code_Submission_Quality_Criteria#Commits

**Pull request closed without merging**